src/random.cpp

```diff
@@ -191,6 +191,7 @@ void GetDevURandom(unsigned char *ent32)
     do {
         ssize_t n = read(f, ent32 + have, NUM_OS_RANDOM_BYTES - have);
         if (n <= 0 || n + have > NUM_OS_RANDOM_BYTES) {
+            close(f);
```

Comment:
As I understand it, most OS's will close descriptors on exit, but it is not guaranteed by the C language, so best practice is to close properly.